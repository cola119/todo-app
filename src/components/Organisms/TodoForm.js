import React, { useState } from 'react';

import TextForm from '../Molecules/Forms/TextForm';
import RadioForm from '../Molecules/Forms/RadioForm';

import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';

const TodoForm = (props) => {


    const [todo, setTodo] = useState(props.formInitial);

    const onSubmit = (e) => {
        e.preventDefault();
        props.onSubmit(todo);
        setTodo({ ...todo, title: '' });
    }

    return (
        <form onSubmit={onSubmit}>
            <Grid
                container
                direction="row"
                justify="center"
                alignItems="center"
            >
                <Grid item xs={12}>
                    <TextForm
                        label={props.label}
                        placeholder={props.placeholder}
                        onChange={e => setTodo({ ...todo, title: e.target.value })}
                        value={todo.title}
                    />
                </Grid>
                <Grid item xs={12} md={6}>
                    <RadioForm
                        legend="期限"
                        labels={Object.entries(props.deadlineLabels)}
                        value={todo.deadline}
                        onChange={e => setTodo({ ...todo, deadline: e.target.value })}
                    />
                </Grid>
                <Grid item xs={12} md={6}>
                    <RadioForm
                        legend="優先度"
                        labels={Object.entries(props.priorityLabels)}
                        value={todo.priority}
                        onChange={e => setTodo({ ...todo, priority: e.target.value })}
                    />
                </Grid>
                <Grid item xs={12}>
                    <Button fullWidth variant="contained" color="primary" type="submit">add</Button>
                </Grid>
            </Grid>
        </form>
    );
};

export default TodoForm;