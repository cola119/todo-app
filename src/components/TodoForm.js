import React, { useState } from 'react';

import TextForm from './Forms/TextForm';
import RadioForm from './Forms/RadioForm';
import Grid from '@material-ui/core/Grid';

import Button from '@material-ui/core/Button';

const TodoForm = (props) => {

    const priorityLabels = {
        high: "高い",
        medium: "普通",
        low: "低い",
    }

    const initial = { title: '', priority: 'medium' };

    const [todo, setTodo] = useState(initial);

    const onSubmit = (e) => {
        e.preventDefault();
        props.onSubmit(todo);
        setTodo(initial);
    }

    return (
        <form onSubmit={onSubmit}>
            <Grid
                container
                // spacing={16}
                direction="row"
                justify="center"
                alignItems="center"
            >
                <Grid item xs={12} sm={8}>
                    <TextForm
                        label={props.label}
                        placeholder={props.placeholder}
                        onChange={e => setTodo({ ...todo, title: e.target.value })}
                        value={todo.title}
                    />
                </Grid>
                <Grid item xs={12} sm={8}>
                    <RadioForm
                        legend="優先度"
                        labels={Object.entries(priorityLabels)}
                        value={todo.priority}
                        onChange={e => setTodo({ ...todo, priority: e.target.value })}
                    />
                </Grid>
                <Grid item xs={12} sm={8}>
                    <Button fullWidth variant="contained" color="primary" type="submit">add</Button>
                </Grid>
            </Grid>
        </form>
    );
};

export default TodoForm;