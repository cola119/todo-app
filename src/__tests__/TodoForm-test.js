import React from 'react';
import renderer from 'react-test-renderer';
import TodoForm from '../components/TodoForm';

test('<TodoForm />のスナップショット', () => {
    const testLabels = {
        hoge: "hogehoge",
        foo: "foofoo",
    }
    const tree = renderer.create(<TodoForm label="label" placeholder="placeholder" deadlineLabels={testLabels} />).toJSON();
    expect(tree).toMatchSnapshot();
})