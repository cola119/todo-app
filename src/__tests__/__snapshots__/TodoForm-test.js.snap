// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm />のスナップショット 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-center-18"
  >
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <div
        className="MuiFormControl-root-98 MuiFormControl-marginNormal-99 MuiFormControl-fullWidth-101"
      >
        <label
          className="MuiFormLabel-root-113 MuiFormLabel-required-118 MuiInputLabel-required-106 MuiInputLabel-root-102 MuiInputLabel-formControl-107 MuiInputLabel-animated-110 MuiInputLabel-shrink-109 MuiInputLabel-outlined-112"
          data-shrink={true}
        >
          label
          <span
            className="MuiFormLabel-asterisk-119"
          >
             *
          </span>
        </label>
        <div
          className="MuiInputBase-root-133 MuiOutlinedInput-root-120 MuiInputBase-fullWidth-142 MuiInputBase-formControl-134"
          onClick={[Function]}
        >
          <fieldset
            aria-hidden={true}
            className="MuiPrivateNotchedOutline-root-150 MuiOutlinedInput-notchedOutline-127"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="MuiPrivateNotchedOutline-legend-151"
              style={
                Object {
                  "width": 0,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
          <input
            aria-invalid={false}
            autoFocus={true}
            className="MuiInputBase-input-143 MuiOutlinedInput-input-128"
            disabled={false}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="placeholder"
            required={true}
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <fieldset
        className="MuiFormControl-root-98"
      >
        <legend
          className="MuiFormLabel-root-113"
        >
          期限
        </legend>
        <div
          aria-label="期限"
          className="MuiFormGroup-root-152 MuiFormGroup-row-153"
          role="radiogroup"
        >
          <label
            className="MuiFormControlLabel-root-154"
          >
            <span
              className="MuiButtonBase-root-175 MuiIconButton-root-169 MuiPrivateSwitchBase-root-165 MuiRadio-root-160 MuiRadio-colorPrimary-163"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-174"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-178"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-168"
                  disabled={false}
                  name="期限"
                  onChange={[Function]}
                  type="radio"
                  value="hoge"
                />
              </span>
              <span
                className="MuiTouchRipple-root-249"
              />
            </span>
            <span
              className="MuiTypography-root-187 MuiTypography-body1-196 MuiFormControlLabel-label-159"
            >
              hogehoge
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-154"
          >
            <span
              className="MuiButtonBase-root-175 MuiIconButton-root-169 MuiPrivateSwitchBase-root-165 MuiRadio-root-160 MuiRadio-colorPrimary-163"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-174"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-178"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-168"
                  disabled={false}
                  name="期限"
                  onChange={[Function]}
                  type="radio"
                  value="foo"
                />
              </span>
              <span
                className="MuiTouchRipple-root-249"
              />
            </span>
            <span
              className="MuiTypography-root-187 MuiTypography-body1-196 MuiFormControlLabel-label-159"
            >
              foofoo
            </span>
          </label>
        </div>
      </fieldset>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <fieldset
        className="MuiFormControl-root-98"
      >
        <legend
          className="MuiFormLabel-root-113"
        >
          優先度
        </legend>
        <div
          aria-label="優先度"
          className="MuiFormGroup-root-152 MuiFormGroup-row-153"
          role="radiogroup"
        >
          <label
            className="MuiFormControlLabel-root-154"
          >
            <span
              className="MuiButtonBase-root-175 MuiIconButton-root-169 MuiPrivateSwitchBase-root-165 MuiRadio-root-160 MuiRadio-colorPrimary-163"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-174"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-178"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-168"
                  disabled={false}
                  name="優先度"
                  onChange={[Function]}
                  type="radio"
                  value="high"
                />
              </span>
              <span
                className="MuiTouchRipple-root-249"
              />
            </span>
            <span
              className="MuiTypography-root-187 MuiTypography-body1-196 MuiFormControlLabel-label-159"
            >
              高い
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-154"
          >
            <span
              className="MuiButtonBase-root-175 MuiIconButton-root-169 MuiPrivateSwitchBase-root-165 MuiRadio-root-160 MuiRadio-colorPrimary-163 MuiPrivateSwitchBase-checked-166 MuiRadio-checked-161"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-174"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-178"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zm0-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={true}
                  className="MuiPrivateSwitchBase-input-168"
                  disabled={false}
                  name="優先度"
                  onChange={[Function]}
                  type="radio"
                  value="medium"
                />
              </span>
              <span
                className="MuiTouchRipple-root-249"
              />
            </span>
            <span
              className="MuiTypography-root-187 MuiTypography-body1-196 MuiFormControlLabel-label-159"
            >
              普通
            </span>
          </label>
          <label
            className="MuiFormControlLabel-root-154"
          >
            <span
              className="MuiButtonBase-root-175 MuiIconButton-root-169 MuiPrivateSwitchBase-root-165 MuiRadio-root-160 MuiRadio-colorPrimary-163"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={null}
            >
              <span
                className="MuiIconButton-label-174"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-178"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                  />
                </svg>
                <input
                  checked={false}
                  className="MuiPrivateSwitchBase-input-168"
                  disabled={false}
                  name="優先度"
                  onChange={[Function]}
                  type="radio"
                  value="low"
                />
              </span>
              <span
                className="MuiTouchRipple-root-249"
              />
            </span>
            <span
              className="MuiTypography-root-187 MuiTypography-body1-196 MuiFormControlLabel-label-159"
            >
              低い
            </span>
          </label>
        </div>
      </fieldset>
    </div>
    <div
      className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
    >
      <button
        className="MuiButtonBase-root-175 MuiButton-root-223 MuiButton-contained-234 MuiButton-containedPrimary-235 MuiButton-raised-237 MuiButton-raisedPrimary-238 MuiButton-fullWidth-248"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-224"
        >
          add
        </span>
        <span
          className="MuiTouchRipple-root-249"
        />
      </button>
    </div>
  </div>
</form>
`;
